Dev console merge with console/master

christianvogt opened the pull request.

spadgett committed:
Add RBAC checks to dialog links

Commit:
Use MultiListPage for BaremetalHostsPage

This allows to use machines and nodes data in hosts page

rawagner committed:
Add details card.
Connect dashboard cards to K8s resources with Firehose

Commit:
Add Nodes to Host MultiListPage

vojtechszocs committed:
Add plugin integration tests

kyoto committed:
Query Browser: Add page Actions menu and kebab menu for each query

kyoto committed:
Query Browser: Make the Hide Graph button also hide the graph controls

Move the "Hide Graph" / "Show Graph" button outside of the graph area
and change it to also hide the graph controls, not just the graph
itself.

jhadvig committed:
Dont link to ImageStreamTag if there is no tag in ImageStream status

Commit:
Convert Baremetal Hosts list to use PF4 table

rhamilto committed:
Enable underline on hover of Import YAML

Commit:
Disable react/prop-types eslint rule

due to false positives in typescript when using React.FC

alecmerdler committed:
include OwnNamespace install mode support for OLM

rhamilto committed:
Remove "Edit Environment" from kebab/actions dropdowns

See https://github.com/openshift/console/pull/1694#issuecomment-500974029

christianvogt committed:
add action item feature to dropdown

christianvogt committed:
add knative icon

christianvogt committed:
update eslint rules

christianvogt committed:
add perspective header to main nav

christianvogt committed:
support redirect to perspective landing page

christianvogt committed:
update vscode settings for tsdk and debug.node.autoAttach

christianvogt committed:
update types for page extensions

christianvogt committed:
change Administrator perspective icon to PF CogsIcon

christianvogt committed:
support children in namespace bar

christianvogt committed:
support namespace redirect for any URL that supports namespaces

christianvogt committed:
add active application redux state, action, reducer

christianvogt committed:
update ownerReference types

christianvogt committed:
export routes#getRouteWebURL

christianvogt committed:
export create-secret#SSHAuthSubform

christianvogt committed:
add dev-console dependencies

christianvogt committed:
contribute dev-console plugin

openshift-merge-robot committed:
Merge pull request #1717 from alecmerdler/OLM-1112

Fix OLM InstallMode Logic

openshift-merge-robot committed:
Merge pull request #1718 from rhamilto/remove-actions

Remove "Edit Environment" from kebab/actions dropdowns

mareklibra committed:
Firehose: honor `optional` prop for watches on individual resources (#1711)

* Firehose: introduce "ignoreIfMissing" resource property

If set to `true`, a resource is skipped from generic
`loadError` 404 checking (cross-resources). So the overall data
retrieval does not fail if a specified resource is missing.

Can be used to optionally query a resource but enable rendering
of children if it is missing (HTTP 404).

The new property is an addition to existing `optional` property which
does not block rendering of children if list-retrieval is still in progress.

With default value `ignoreIfMissing = false`, the component is
backward compatible.

* Firehose: fix the `optional` property for 404

If a resource is `optional: true` then it is excluded from
setting of top-level loadError.

Firehose's child component can still check loadError by
inspecting the resource directly.

vikram-raj commented:
@christianvogt  I am getting this when I run this PR locally.
```
template: index.html:8:10: executing "index.html" at <.CustomProductName>: can't evaluate field CustomProductName in type *server.jsGlobals
```

And in terminal this
```
2019/06/14 07:10:56 http: multiple response.WriteHeader calls
```

christianvogt commented:
@vikram-raj console has changed a lot including the server. Make sure you rebuild using `./build.sh`

openshift-merge-robot committed:
Merge pull request #1715 from rhamilto/import-yaml-hover

Enable underline on hover of Import YAML

vikram-raj commented:
1. Application dropdown is not opening it is throwing an error.
2. Git Import page takes 15 -18 sec to load. and also no loader is visible just blank page is visible.
3. Getting this warning in the console
```
checkPropTypes.js:19 Warning: Failed prop type: Invalid prop `path` of type `array` supplied to `Route`, expected `string`.
    in Route
```
4. Environment variable field for ```Build Config``` and ```Deployment Config``` is in read-only mode. Not able to add variables.

vikram-raj commented:
Do we need ```Import YAML``` action in the secondary navbar of DevConsole as we already have ```Import YAML``` option on add page?

christianvogt commented:
@vikram-raj 
1. App drop down opens for me. But it is showing the same app in all namespaces and I cannot select it.... I'll have to look into this and compare to our current fork.
2. Loads ok for me. But i can see the blank page for a little bit.
3. This is a react-router warning. I'll check if this is something that is happening because of my work or not. I know that the warning is false because it does support an array prop. Console tries to suppress the error with a hack in `app.jsx`
4. Odd it's set to `readOnly={false}`

The Import YAML button is in admin console and that is a shared component. @serenamarie125 What do you want to do about this button?

sbose78 commented:
build and deploy

sbose78 commented:
Your build is progress : https://console-openshift-console.apps.rohit37.devcluster.openshift.com/k8s/ns/openshift-console/builds
 
 PR will be LIVE in https://pr-140-openshift-console.apps.rohit37.devcluster.openshift.com

sahil143 reviewed (commented):
Base Circle Svg is not showing up on any of the nodes and Decorator. DropShadow filter is not showing on hover.
![Screenshot from 2019-06-14 11-33-22](https://user-images.githubusercontent.com/9278015/59486970-c43cc180-8e98-11e9-8c15-081da2b44945.png)
![Screenshot from 2019-06-14 11-33-06](https://user-images.githubusercontent.com/9278015/59486863-6f994680-8e98-11e9-91dd-df6095b5ee98.png)

Tooltips and application name don't have black background.
![Screenshot from 2019-06-14 11-22-54](https://user-images.githubusercontent.com/9278015/59487082-331a1a80-8e99-11e9-8451-1bb01508ed96.png)

By default, Topology is not selected on side bar.
![Screenshot from 2019-06-14 11-23-11](https://user-images.githubusercontent.com/9278015/59487217-91df9400-8e99-11e9-8300-30fa038238e6.png)

Selected Item doesn't have Blue Vertical line on the left side.
![Screenshot from 2019-06-14 11-23-36](https://user-images.githubusercontent.com/9278015/59487290-d79c5c80-8e99-11e9-8a51-f2ccd9f48d05.png)

sahil143 commented:
Clicking on Filter by Application Dropdown on Topology page throws an error
```
Uncaught Invariant Violation: Objects are not valid as a React child (found: object with keys {name}). If you meant to render a collection of children, use an array instead.
    in a (created by DropDownRow)
    in li (created by DropDownRow)
    in DropDownRow (created by Dropdown)
    in ul (created by Dropdown)
    in div (created by Dropdown)
    in div (created by Dropdown)
    in Dropdown (created by ResourceDropdown)
    in ResourceDropdown (created by ApplicationDropdown)
```
Also, I'm not getting the dropdown to select the `part-of` application while creating S2I app from Browse Catalog instead there is an input field to enter the labels.

divyanshiGupta reviewed (commented):
![nav-bug](https://user-images.githubusercontent.com/20724543/59494049-f6a2ea80-8ea9-11e9-9c10-2056be17bfbb.gif)
So when I click on `projects` I can see the list of all projects then I click on `status` and I see the status page, but `projects` remains selected and on clicking outside `status` gets deselected but `projects` remain selected even though we are on the `status` page

Is this bug known or it is in this PR or something has changed in console?

Also, it used to show `overview` in the URL and not `projects` previously, and it is also getting redirected to `workloads`.

rohitkrai03 commented on frontend/packages/dev-console/src/plugin.tsx:
```diff
@@ -0,0 +1,208 @@
+    },
+  },
+  {
+    type: 'NavItem/Href',
```
I think we don't need `Status` nav item anymore because console moved this page into projects details page as workloads section.

christianvogt replied:
Removed

rohitkrai03 commented:
@christianvogt I found a fix for the environment fields being readonly. It was because `environment.jsx` implementation changed a bit. I'll raise a PR to your branch.

rohitkrai03 commented:
@christianvogt Raised a PR on your branch for some of the fixes - https://github.com/christianvogt/console/pull/1

debsmita1 commented:
> @christianvogt I am getting this when I run this PR locally.
> 
> ```
> template: index.html:8:10: executing "index.html" at <.CustomProductName>: can't evaluate field CustomProductName in type *server.jsGlobals
> ```
> 
> And in terminal this
> 
> ```
> 2019/06/14 07:10:56 http: multiple response.WriteHeader calls
> ```

@christianvogt I am getting this error. Tried to follow what you suggested to Vikram, but in vain

openshift-merge-robot committed:
Merge pull request #1720 from spadgett/rbac-modals

Add RBAC checks to dialog links

rhamilto committed:
Add standard `Resource > Resource Details` breadcrumb to resource details pages

Fixes https://jira.coreos.com/browse/CONSOLE-1539

spadgett committed:
Bug 1720119: Tighten node IP query

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1720119

TheRealJon committed:
Improve area and bar chart readability

christianvogt committed:
support highlighting active root nav items outside of a section

karthikjeeyar commented on frontend/packages/dev-console/yamls/pipelines/pipeline.yaml:
```diff
@@ -0,0 +1,115 @@
+        resource: mapit-image
+  - name: deploy
+    taskRef:
+      name: openshift-cli-deploy-mapit
```
```suggestion
      name: openshift-cli
```
Raised a PR for this PR https://github.com/talamer/console/pull/142.

openshift-merge-robot committed:
Merge pull request #1726 from spadgett/node-charts

Bug 1720119: Tighten node IP query

Commit:
Use machine selectors from @console/shared

* update machine selector types to use specific kinds
* remove selectors definition from machine module
* use machine selectors from @console/shared in machine, machine-set
  and machine-deployment modules

spadgett committed:
Fix broken ClusterServiceVersion instance links

`apiVersion` and `kind` weren't properly injected into the object, and
depending on the URL, the ClusterServiceVersion resource name wasn't found.

openshift-merge-robot committed:
Merge pull request #1712 from jhadvig/tag-links

Dont link to ImageStreamTag if there is no tag in ImageStream status

rhamilto committed:
Align <title> with <h1> for "Installed Operators"

alecmerdler committed:
use metadata.name instead of fieldSelector for OLM e2e cleanup

openshift-merge-robot committed:
Merge pull request #1727 from TheRealJon/bug-1719481

Improve area and bar chart readability

vojtechszocs committed:
Add support for custom table filters

openshift-merge-robot committed:
Merge pull request #1730 from spadgett/broken-csv-isntance-link

Fix broken ClusterServiceVersion instance links

openshift-merge-robot committed:
Merge pull request #1734 from alecmerdler/better-olm-cleanup

Better OLM E2E Cleanup

vojtechszocs committed:
Reuse existing rowFilters prop to contain filter functions

vojtechszocs committed:
Ensure sensible rowFilters default

vojtechszocs committed:
Fix the ordering in getAllTableFilters

openshift-merge-robot committed:
Merge pull request #1710 from kyoto/query-browser-move-hide-button

Query Browser: Make the Hide Graph button also hide the graph controls

openshift-merge-robot committed:
Merge pull request #1733 from rhamilto/csv-title-fix

Align <title> with <h1> for "Installed Operators"

openshift-merge-robot committed:
Merge pull request #1698 from jtomasek/multi-list-page

 Use MultiListPage for BaremetalHostsPage

openshift-merge-robot committed:
Merge pull request #1701 from kyoto/query-browser-delete-query

Query Browser: Add page Actions menu and kebab menu for each query

Commit:
Add unit test for limitRange

mareklibra committed:
Upgrade kubevirt-web-ui-components to v0.1.35

openshift-merge-robot committed:
Merge pull request #1735 from vojtechszocs/custom-table-filters

Add support for custom table filters

vojtechszocs committed:
Jest config updates

vojtechszocs committed:
Jest config updates

vojtechszocs committed:
Fix error in Table stateToProps

openshift-merge-robot committed:
Merge pull request #1725 from rhamilto/breadcrumb-changes

Add standard `Resource > Resource Details` breadcrumb to resource det…

spadgett committed:
Remove misleading route hostname message

The create route form says a route hostname can't be changed. This is
not always the case and depends on the user's permissions. Remove the
message.

rhamilto committed:
Fix bug where error occurred when clicking Update Channel Save without interacting with channel dropdown

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1721431

alecmerdler committed:
improve feature detection for OLM nav items on upstream k8s

spadgett committed:
Make cluster status default page for native k8s

spadgett committed:
Limit alert height in modal editors

openshift-merge-robot committed:
Merge pull request #1738 from mareklibra/upgradeComponents.v0.1.35

Upgrade kubevirt-web-ui-components to v0.1.35

openshift-merge-robot committed:
Merge pull request #1743 from vojtechszocs/jest-config-updates

Jest config updates

openshift-merge-robot committed:
Merge pull request #1744 from vojtechszocs/fix-filter

Fix error in Table stateToProps

openshift-merge-robot committed:
Merge pull request #1746 from rhamilto/bz-1721431

Bug 1721431:  Fix bug where error occurred when clicking Update Channel Save withou…

openshift-merge-robot committed:
Merge pull request #1747 from alecmerdler/CONSOLE-1544

Improve Feature Detection for OLM Nav Items

openshift-merge-robot committed:
Merge pull request #1745 from spadgett/fix-hostname-message

Remove misleading route hostname message

openshift-merge-robot committed:
Merge pull request #1749 from spadgett/modal-alert-height

Limit alert height in modal editors

openshift-merge-robot committed:
Merge pull request #1696 from rawagner/dashboard-k8s-simple

Add details card

openshift-merge-robot committed:
Merge pull request #1748 from spadgett/native-k8s-default-page

Make cluster status default page for native k8s

openshift-merge-robot committed:
Merge pull request #1737 from xiaocwan/limitrangeUnitTest

Add unit test for limitRange

openshift-merge-robot committed:
Merge pull request #1708 from vojtechszocs/plugin-tests

Add plugin extension tests

yaacov committed:
Use plural instead of path (#1728)

* use plural instead of path

* change alert path to plural

christianvogt committed:
enable dev-console plugin in console-app
